swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift

```diff
@@ -69,6 +59,7 @@ public final class SessionViewModel: ObservableObject {
 @MainActor
 struct SessionView: View {
   @ObservedObject var model: SessionViewModel
+  @EnvironmentObject var favorites: Favorites
```

Comment:
One thing to be mindful here is that by introducing multiple dependencies into your view, this will be more difficult to test. If a view's only dependency is its viewmodel and you don't "reach into" the model with `model.` but just call functions on it, it would be possible to write unit-tests against the viewmodel and assert how its properties change (which in turn should be the only thing driving the state of the view).

Reply:
> you don't "reach into" the model with model

This is the whole point of `@Published var favorites: Favorites` defined in the model - in the view, we observe the model and then we _should_ automatically receive updates to its properties.

The issue is that if we're simply reading state and don't need to interact with it, passing directly to the view via dependency injection is much less boilerplate since we save having to thread all of the view's needed readonly properties through the model.

Swift has `@EnvironmentObject` expressly for this purpose.

It's not always clear which one to reach for versus another, but I'm open to further discussion.

Reply:
When passing a shared object like `Favorites` that needs to be referenced in many views like Settings, SessionView, etc, the two options are:

1. Thread it through each viewmodel in the view hierarchy initializer, using `var favorites: Favorites` in each model in order to pass it on to the next model. This means lots of potential boilerplate to get the favorites to actual view that needs it.
2. Declare it as an `@EnvironmentObject` in the root `FirezoneApp.swift` and then dependency inject it directly into the view that needs it.

One instance of `favorites` is preferred so that, for example, clearing them in settings resets the SessionView right away.

Reply:
I think the takeaway in this case is `Favorites` is an `ObservableObject` so that it can be dependency-inject directly in the view. In other words, it's a viewmodel, just not by name.

We either need to remove its observable properties and access it solely through the model, or we should relax our requirement that `model` is the only observable state in views as a convention.

Either option above wouldn't change the fact that the view ultimately relies on `favorites` to be there - either within the `model` or as its own quasi-model.

Reply:
Yeah, it seems that Swift doesn't really encourage this programming style. It is a trade-off and given that we currently don't have any tests for this stuff anyway, we might as well use the paradigm that is the least boilerplate